Cython/Compiler/TypeStubGenerator.py

```diff
@@ -0,0 +1,425 @@
+from .Compiler import Version
+from .Compiler.Nodes import CNameDeclaratorNode
+from .Compiler.ExprNodes import CallNode, NameNode, ImportNode, TupleNode, AttributeNode
+from .CodeWriter import DeclarationWriter
+from .Compiler.ParseTreeTransforms import CythonTransform
+from .Compiler import PyrexTypes
+from .Compiler.Main import Context
```

Comment:
These relative imports look wrong. We're in the `Cython.Compiler` package already.
```suggestion
from . import Version
from .Nodes import CNameDeclaratorNode
from .ExprNodes import CallNode, NameNode, ImportNode, TupleNode, AttributeNode
from ..CodeWriter import DeclarationWriter
from .ParseTreeTransforms import CythonTransform
from . import PyrexTypes
```
Please don't import `Main` because it imports everything and thus necessarily leads to circular imports.
I'd also avoid importing names directly from `Nodes` and `ExprNodes` because they may also suffer from circular imports. Just import the module and use `ExprNodes.NameNode` etc.